### ISSUE TITLE
Client not brpoping after lost/re-establishing connection

### ISSUE THREAD

**Issue by klinquist** [luin/ioredis#22]
I noticed that after I restarted redis-server, none of my services (all listening on lists) were reading from their lists.  I was able to duplicate the problem this morning with this code:
```
var Redis = require('ioredis');
var client = new Redis();

function ListenToRedis(){
	client.brpop('testlist', 0, function(listname, item){
		console.log(item);
		process.nextTick(function(){
			ListenToRedis();
		});
	});
	
}
ListenToRedis();
```

If I restart my redis server (even very quickly - sudo service redis-server restart), items begin building up in the list, the above script is no longer brpoping them.  How can I solve this?

Thanks!

**Comment by luin** [comment]
This behavior is currently by designed since auto re-brpop may have side effects. For instance when you give brpop a timeout other than 0, it would be ambiguity to decide the value of the timeout for the new brpop when reconnected.

**Comment by luin** [comment]
You may send the brpop command on the `ready` event which will be emitted when connected/reconnected:
```javascript
var client = new Redis();
client.on('ready', ListenToRedis);
```

However I'm considering to add an option to enable auto re-brpop when reconnected.

**Comment by klinquist** [comment]
I would love that option!  Thank you for the reply.

**Comment by klinquist** [comment]
You rock! Thank you!

**Comment by luin** [comment]
Released in 1.2.0. Doc: https://github.com/luin/ioredis#auto-reconnect
